File: scripts/create-promo.js
#!/usr/bin/env node
/**
 * Создание промокодов
 * Использование: node scripts/create-promo.js CODE type value min_amount [max_uses] [expires_days]
 */

const db = require('../database');
db.initializeDatabase();

const [,, code, type, value, minAmount, maxUses, expiresDays] = process.argv;

if (!code || !type || !value || !minAmount) {
    console.log(`
Использование: node scripts/create-promo.js <CODE> <type> <value> <min_amount> [max_uses] [expires_days]

Примеры:
  node scripts/create-promo.js FIRST10 percent 10 500
  node scripts/create-promo.js SALE500 fixed 500 2000 100 30

Параметры:
  CODE        - Код промокода
  type        - percent | fixed
  value       - Размер скидки
  min_amount  - Минимальная сумма заказа
  max_uses    - Макс. использований (опционально)
  expires_days - Срок действия в днях (опционально)
`);
    process.exit(1);
}

if (!['percent', 'fixed'].includes(type)) {
    console.error('❌ type должен быть "percent" или "fixed"');
    process.exit(1);
}

try {
    const data = {
        code: code.toUpperCase(),
        discountType: type,
        discountValue: parseFloat(value),
        minOrderAmount: parseFloat(minAmount),
        maxUses: maxUses ? parseInt(maxUses) : null,
        expiresAt: expiresDays ? new Date(Date.now() + parseInt(expiresDays) * 86400000).toISOString() : null
    };

    db.createPromoCode(data);

    console.log(`✅ Промокод создан!
  Код: ${data.code}
  Тип: ${type === 'percent' ? 'Процент' : 'Фиксированная'}
  Скидка: ${type === 'percent' ? value + '%' : value + '₽'}
  Мин. сумма: ${minAmount}₽${data.maxUses ? `\n  Макс. использований: ${data.maxUses}` : ''}${data.expiresAt ? `\n  Истекает: ${new Date(data.expiresAt).toLocaleDateString('ru-RU')}` : ''}`);
} catch (e) {
    console.error(e.message.includes('UNIQUE') ? '❌ Такой промокод уже существует' : `❌ Ошибка: ${e.message}`);
    process.exit(1);
}

File: scripts/create_promo.py
import asyncio
import sys
import os
from datetime import datetime, timedelta

# Add project root to path
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from app.db.database import AsyncSessionLocal
from app.db.models import PromoCode

async def create_promo():
    if len(sys.argv) < 5:
        print("Использование: python scripts/create_promo.py <CODE> <type> <value> <min_amount> [max_uses] [expires_days]")
        print("Пример: python scripts/create_promo.py SUMMER2024 percent 10 1000")
        return

    code = sys.argv[1].upper()
    discount_type = sys.argv[2]
    try:
        discount_value = float(sys.argv[3])
        min_amount = float(sys.argv[4])
        max_uses = int(sys.argv[5]) if len(sys.argv) > 5 else None
        expires_days = int(sys.argv[6]) if len(sys.argv) > 6 else None
    except ValueError:
        print("Ошибка: Неверный формат чисел")
        return

    if discount_type not in ['percent', 'fixed']:
        print("Ошибка: type должен быть 'percent' или 'fixed'")
        return

    expires_at = datetime.now() + timedelta(days=expires_days) if expires_days else None

    async with AsyncSessionLocal() as session:
        promo = PromoCode(
            code=code,
            discount_type=discount_type,
            discount_value=discount_value,
            min_order_amount=min_amount,
            max_uses=max_uses,
            expires_at=expires_at
        )
        session.add(promo)
        try:
            await session.commit()
            print(f"✅ Промокод {code} создан!")
        except Exception as e:
            print(f"❌ Ошибка создания: {e}")

if __name__ == "__main__":
    asyncio.run(create_promo())

